### PR TITLE
Article 093: Rice's Theorem

### PR DESCRIPTION
**Pull request by s3lph**
#### Length of text ####
:book:  long

####  Difficulty to read ####
:elephant:  heavy

####  Difficulty to understand #### 
:mortar_board:  difficult

####  Type of text #### 
:bowtie:  informative

####  Type of header #### 
:boom:  catchy

### CODE REVIEW

**Comment by Teifun2** [comment]
**Plenum:**
Problems:
- Remove Theoreme from Title

Ideas:
- undecidable in general how to explain?
- Better Example? Bugs?

**Comment by Teifun2** [comment]
- Image missing otherwise good

**Comment by MoiraZuber** [comment]
Added an image suggestion.

File: articles/article-093-rices-theorem.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<!-- BEGIN TITLE -->
+# Why There Is No Perfect Anti-Virus
+<!-- END TITLE -->
+
+<!-- BEGIN BODY -->
+In computer science, simple "Yes/No" questions are called **decision
+problems**.  A decision problem is called **decidable** if we can
+write a program that can always find an answer for such a problem.
+
+An example for such a decision problem is "does this program contain a
```
Comment (noverney) [inline articles/article-093-rices-theorem.md]: I would add a question mark to the "does this program contain a virus?", for clarity

File: articles/article-093-rices-theorem.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+computer virus".  If this problem was decidable, there could be
+another program that could always correctly answer this question with
+"Yes" or "No".  A mathematician called Henry Gordon Rice showed that
+such decision problems about the behavior of another program are
+always undecidable, unless the answer is **always** the same.
+
+As most viruses are hidden inside other software and do not always
+expose their real nature, this decision problem can't be solved
```
Comment (noverney) [inline articles/article-093-rices-theorem.md]: As most viruses are hidden inside other software and don't always expose their true intention, they are difficult to decide using Rice's theorem.

File: articles/article-093-rices-theorem.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+<!-- BEGIN BODY -->
+In computer science, simple "Yes/No" questions are called **decision
+problems**.  A decision problem is called **decidable** if we can
+write a program that can always find an answer for such a problem.
+
+An example for such a decision problem is "does this program contain a
+computer virus".  If this problem was decidable, there could be
+another program that could always correctly answer this question with
+"Yes" or "No".  A mathematician called Henry Gordon Rice showed that
```
Comment (noverney) [inline articles/article-093-rices-theorem.md]: with a "yes" or "no". 
You do not need to capitalize the yes and no (you can make the change above if you want).

Comment (noverney) [inline articles/article-093-rices-theorem.md]: usually you would say:
A mathematician named Henry...

Comment (s3lph) [reply articles/article-093-rices-theorem.md]: Ok

Comment (s3lph) [reply articles/article-093-rices-theorem.md]: Agreed

File: articles/article-093-rices-theorem.md (hunk 1 above)
Comment (s3lph) [reply articles/article-093-rices-theorem.md]: Agreed

File: articles/article-093-rices-theorem.md (hunk 2 above)
Comment (s3lph) [reply articles/article-093-rices-theorem.md]: I agree with most of your proposed changes, except for the last part: you don't decide a virus, and you don't do it using Rice's theorem.

**Pull request merged**